### DISCUSSION TITLE
OPENSSL Issue: Failed to load OpenSSL legacy provider

### DISCUSSION THREAD

**Discussion by Pruthwiraj-Kumar-Deo** [openssl/openssl#21417] [Q&A] [answered]
Hi Team
I am currently cross compiling my binary with OPENSSL (3.1.1) and POCO (1.11.6) Library .

I have built my OPENSSL library with 'no-asm no-idea no-mdc2 no-rc5 no-ssl3 enable-weak-ssl-ciphers' flags.
and built POCO library with '--poquito --no-tests --no-samples --omit=CppUnit,CppUnit/WinTestRunner,XML,JSON,Data,Data/SQLite,Data/ODBC,Data/MySQL,MongoDB,Zip,Encodings,Redis --static' flags.

Then I built my binary using above two libraries. But while running the binary I am getting an exception :

Crypto Exception: Failed to load OpenSSL legacy provider

When I tracked it down , found that it is coming from 
src/Crypto/src/OpenSSLInitializer.cpp:140: if (!_legacyProvider) throw CryptoException("Failed to load OpenSSL legacy provider");

So I tried to provide the legacy.so in /usr/lib during run time as well I set 
openssl_conf = openssl_init

[openssl_init]
providers = provider_sect

[provider_sect]
default = default_sect
legacy = legacy_sect

[legacy_sect]
activate = 1

in /etc/ssl/openssl.cnf also. But still facing the same issue. So, I have some query about it.

1.	Why is the exception reoccurring even if I provide the legacy.so at runtime ?
Should I write a logic to load the legacy.so at runtime additionally?
2.	Are OPENSSL (3.1.1) and POCO (1.11.6) Libraries compactible to each other ?
Or should I go for any other version combination ?
Is there any way we can know about the compactible versions of OPENSSL & POCO with each other ?
3.	Any other  library (other than legacy.so )is required  ?

**Comment by t8m** [comment, marked as answer]
The legacy.so must be placed in a directory where the libcrypto looks for it. `make install` should place it there.
